pkg/api/repository.go

```diff
@@ -13,18 +13,38 @@ type Repository struct {
 	OrgID               string `json:"org_id" readonly:"true"`             //Organization ID of the owner
 }
 
-// CreateRepository holds data received from request to create repository
-type CreateRepository struct {
-	UUID                string `json:"uuid" readonly:"true" swaggerignore:"true"`
-	Name                string `json:"name"`
-	URL                 string `json:"url"`                                             //URL of the remote yum repository
-	DistributionVersion string `json:"distribution_version" example:"7"`                //Version to restrict client usage to
-	DistributionArch    string `json:"distribution_arch" example:"x86_64"`              //Architecture to restrict client usage to
-	AccountID           string `json:"account_id" readonly:"true" swaggerignore:"true"` //Account ID of the owner
-	OrgID               string `json:"org_id" readonly:"true" swaggerignore:"true"`     //Organization ID of the owner
+// RepositoryRequest holds data received from request to create repository
+type RepositoryRequest struct {
+	UUID                *string `json:"uuid" readonly:"true" swaggerignore:"true"`
+	Name                *string `json:"name"`
+	URL                 *string `json:"url"`                                             //URL of the remote yum repository
+	DistributionVersion *string `json:"distribution_version" example:"7"`                //Version to restrict client usage to
+	DistributionArch    *string `json:"distribution_arch" example:"x86_64"`              //Architecture to restrict client usage to
+	AccountID           *string `json:"account_id" readonly:"true" swaggerignore:"true"` //Account ID of the owner
+	OrgID               *string `json:"org_id" readonly:"true" swaggerignore:"true"`     //Organization ID of the owner
+}
+
+func (r *RepositoryRequest) FillDefaults() {
+	//Fill in default values in case of PUT request, doesn't have to be valid, let the db validate that
+	defaultName := ""
+	defaultUrl := ""
+	defaultVersion := ""
+	defaultArch := ""
```

Comment:
Right now the code isn't blocking "" URLs or "" Names. This might be the place to fix it?

Reply:
I thought the model might be a better place? but i'm not sure.  I'll file a bug around it and we can take a look

Reply:
filed!